rust/agama-lib/src/security/client.rs

```diff
@@ -56,7 +58,7 @@ impl SecurityClient<'_> {
     ) -> Result<(), ServiceError> {
         let dbus_list: Vec<(&str, &str)> = list
             .iter()
-            .map(|s| (s.algorithm.to_str(), s.fingerprint.as_str()))
+            .map(|s| (s.algorithm.clone().into(), s.fingerprint.as_str()))
```

Comment:
You can save the explicit clone if you derive `Copy`. In an enum, it makes sense.